### PR TITLE
Add macOS support

### PR DESCRIPTION
**Pull request by G0V1NDS**
Changes needed for https://github.com/rszyma/kanata-tray/issues/16 building on macos.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -8,6 +8,9 @@ run:
 build_release_linux version="latest":
     GOOS=linux CGO_ENABLED=1 GO111MODULE=on go build -ldflags "-s -w -X 'main.buildVersion={{version}}' -X 'main.buildHash=$(git rev-parse HEAD)' -X 'main.buildDate=$(date -u)'" -trimpath -o dist/kanata-tray
 
+build_release_darwin version="latest":
+    GOOS=darwin GOARCH=amd64 CGO_ENABLED=1 GO111MODULE=on go build -ldflags "-s -w -X 'main.buildVersion=latest' -X 'main.buildHash=$(git rev-parse HEAD)' -X 'main.buildDate=$(date -u)'" -trimpath -o dist/kanata-tray
```
Comment (rszyma) [inline justfile:12]: Please remove `GOARCH=amd64` if we also want to support x86 macs. Without it, a correct architecture will be selected automatically for current platform.

**Review by rszyma — CHANGES_REQUESTED** [review]

**Comment by rszyma** [comment]
1. Does macos require any dependencies? If yes please add them to readme.

2. It would be nice to add macos to CI (see https://github.com/rszyma/kanata-tray/tree/main/.github/workflows)

File: justfile (hunk 1 above)
Comment (G0V1NDS) [reply justfile:12]: Sure, I can do that.

**Comment by G0V1NDS** [comment]
> Does macos require any dependencies? If yes please add them to readme.
I did not require any dependencies so can't commit if any was pre-requisite, will have to try it on a fresh system.

> It would be nice to add macos to CI (see https://github.com/rszyma/kanata-tray/tree/main/.github/workflows)
Let me try adding the workflow, it will help us determine if any dependency is required.

**Comment by rszyma** [comment]
> > Does macos require any dependencies? If yes please add them to readme.
> 
> I did not require any dependencies so can't commit if any was pre-requisite, will have to try it on a fresh system.
> 
> > It would be nice to add macos to CI (see [`main`/.github/workflows](https://github.com/rszyma/kanata-tray/tree/main/.github/workflows?rgh-link-date=2024-04-16T03%3A20%3A10Z))
> 
> Let me try adding the workflow, it will help us determine if any dependency is required.

There won't be any build time dependency, I rather thought about runtime dependency (look at linux dependencies in readme, these are runtime ones).

File: .github/workflows/release.yml
Comment (rszyma) [inline .github/workflows/release.yml:1]: Isn't it possible to cross-compile for macos from ubuntu like it's done for windows? There's would be a lot of duplication after applying the proposed changes.

File: justfile [hunk 2]
```diff
@@ -8,6 +8,9 @@ run:
 build_release_linux version="latest":
     GOOS=linux CGO_ENABLED=1 GO111MODULE=on go build -ldflags "-s -w -X 'main.buildVersion={{version}}' -X 'main.buildHash=$(git rev-parse HEAD)' -X 'main.buildDate=$(date -u)'" -trimpath -o dist/kanata-tray
 
+build_release_darwin version="latest":
+    GOOS=darwin CGO_ENABLED=1 GO111MODULE=on go build -ldflags "-s -w -X 'main.buildVersion=latest' -X 'main.buildHash=$(git rev-parse HEAD)' -X 'main.buildDate=$(date -u)'" -trimpath -o dist/kanata-tray-darwin
```
Comment (rszyma) [inline justfile:12]:
```suggestion
    GOOS=darwin CGO_ENABLED=1 GO111MODULE=on go build -ldflags "-s -w -X 'main.buildVersion={{version}}' -X 'main.buildHash=$(git rev-parse HEAD)' -X 'main.buildDate=$(date -u)'" -trimpath -o dist/kanata-tray-darwin
```

File: .github/workflows/release.yml
Comment (G0V1NDS) [reply .github/workflows/release.yml:1]: I actually never actually built cross-platform builds before so don't have much idea about it. Though I tried running the github actions locally to build it on same ubuntu runner but faced issues related to clang and so I tried the macos runner and it was successful so did this change.

File: justfile (hunk 2 above)
Comment (G0V1NDS) [reply justfile]: Corrected.

**Comment by rszyma** [comment]
I wanted to see the what's the error with cross-compiling, and it looks like it's because used compiler for C stuff is clang instead of expected gcc (I guess) https://github.com/rszyma/kanata-tray/actions/runs/8730443667/job/23954275895

**Pull request merged**